packages/cursorless-engine/src/util/allocateHats/allocateHats.ts

```diff
@@ -135,19 +143,27 @@ function getTokenOldHatMap(oldTokenHats: readonly TokenHat[]) {
 function getTokenRemainingHatCandidates(
   tokenGraphemeSplitter: TokenGraphemeSplitter,
   token: Token,
-  availableGraphemeStyles: DefaultMap<string, HatStyleMap>,
+  availableGraphemeStyles: DefaultMap<string, HatStyleName[]>,
```

Comment:
why not call this one `graphemeRemainingHatCandidates`, as we do in the calling function?

Reply:
deeply ingrained habits from go. these are already extraordinarily long by my usual standards. :P

see https://google.github.io/styleguide/go/decisions#variable-names

anyway, I'm in rome, I have fixed them :)

Reply:
Definitely some interesting heuristics in there. I still err strongly on the side of verbosity to avoid ambiguity, and I also like the visual echoing of incorporating a full name into a bigger name

My biggest gripe with the linked naming convention is that I think omitting words due to context forces you to keep more of the context in your head. I would argue that the cognitive load is higher when one tries to be clever with naming and omit words.

For example, I could see shortening the variable to `graphemeRemainingStyles`, but `hatStyle` is a term of art, and we try to be consistent about using it everywhere. If we sometimes use `style`, and sometimes `hatStyle`, I'd argue that makes the code harder to read

That being said, I do like some of the ideas in there, esp the notion of taking into account length / complexity of variable scope, and the idea that "the name of a local variable should reflect what it contains and how it is being used in the current context, rather than where the value originated"

Cc/ @AndreasArvidsson

Reply:
Wow yeah very different-looking 😅

I def didn't take it as proselytising, no worries 😊